pkg/cache/multi_namespace_cache.go

```diff
@@ -190,7 +190,10 @@ func (c *multiNamespaceCache) List(ctx context.Context, list client.ObjectList,
 	limitSet := listOpts.Limit > 0
 
 	var resourceVersion string
-	for _, cache := range c.namespaceToCache {
+	for ns, cache := range c.namespaceToCache {
```

Comment:
Can we somehow test that after we do a List for a namespaced object (say pods), only the namespaced caches have an informer for that object?

Reply:
actually, nope. The global cache also has an informer for the namespaced object. That seems to be wrong.

Reply:
How about that test? We could maybe also just make the global cache a new attribute in the `multiNamespaceCache` to avoid this exact issue of looping and accidentally including it

Reply:
have modified this. Created a separate clusterCache in `multinamespaceCache`